cwe_checker_rs/src/checkers/cwe_676.rs

```diff
@@ -0,0 +1,108 @@
+/// The CWE 676 checks for potentially dangerous function calls, such as strcpy, memcpy, scanf etc.
+
+use crate::{
+    intermediate_representation::{ Program, Project, Sub, Term, Tid, Jmp, ExternSymbol }, 
+    utils::log::{ CweWarning, LogMessage }};
+use serde::{ Serialize, Deserialize };
+
+const VERSION: &str = "0.1";
+
+pub static CWE_MODULE: crate::CweModule = crate::CweModule {
+    name: "CWE676",
+    version: VERSION,
+    run: check_cwe,
+};
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Hash, Clone)]
+pub struct Config {
+    symbols: Vec<String>,
+}
+
+
+/// For each block of a subroutine check if the jump is a direct call and if the target TID of that call is equal to a dangerous symbol TID
+pub fn get_call_to_target<'a>(caller: &'a Term<Sub>, target: &'a ExternSymbol) -> Vec<Option<(&'a str, &'a Tid, &'a str)>> {
```

Comment:
You can remove the `Option` in the return type, since you only add the CWE hits to the Vec anyway.